vscode-dotnet-runtime-library/src/Utils/ErrorHandler.ts

```diff
@@ -14,22 +14,24 @@ export async function callWithErrorHandling<T>(callback: () => T, context: IIssu
     try {
         return await callback();
     } catch (error) {
-        if ((error.message as string).includes(timeoutConstants.timeoutMessage)) {
-            window.showErrorMessage(`${errorConstants.errorMessage}: ${ error.message }`, timeoutConstants.moreInfoOption).then(async (response: string | undefined) => {
-                if (response === timeoutConstants.moreInfoOption) {
-                    open(timeoutConstants.timeoutInfoUrl);
-                }
-            });
-        } else if (error.constructor.name !== 'UserCancelledError' && showMessage) {
-            window.showErrorMessage(`${errorConstants.errorMessage}: ${ error.message }`, errorConstants.reportOption, errorConstants.hideOption).then(async (response: string | undefined) => {
-                if (response === errorConstants.hideOption) {
-                    showMessage = false;
-                } else if (response === errorConstants.reportOption) {
-                    const [url, issueBody] = formatIssueUrl(error, context);
-                    await env.clipboard.writeText(issueBody);
-                    open(url);
-                }
-            });
+        if (context.errorConfiguration === ErrorConfiguration.DisplayAllErrorPopups) {

```

Comment:
anyway to test these changes?

Reply:
Just pushed an update with tests added, I added a layer of abstraction for displaying popups, as you suggested.